File: python/jzoffer-30.py
# -*- coding:utf-8 -*-
class Solution:
    def FindGreatestSumOfSubArray(self, array):
        # write code here
        if not array:
            return 0
        g = -10000000000
        l = -10000000000
        for n in array:
            g = max(n,n+g)
            l = max(g,l)
        return l

File: java/jzoffer-30.java
public class Solution {
    public int FindGreatestSumOfSubArray(int[] array) {
        if(array == null){
            return 0;
        }
        int g = -10000000;
        int l = -10000000;
        for(int i = 0; i < array.length; i++){
            g = ((array[i] + g) > array[i]) ? (array[i] + g) : array[i];
            l = (g > l) ? g : l;
        }
        return l;
    }
}